packages/eth-deposit/scripts/exec-viaLib.js

```diff
@@ -25,78 +24,67 @@ const ethToL2DepositAmount = parseEther('0.0001')
 
 const main = async () => {
   await arbLog('Deposit Eth via arb-ts')
-  /**
-   * Use arb-ts networks file to retrieve the Rinkeby Inbox address
-   */
 
-   const l1ChainId = await l1Wallet.getChainId()
-   const l1Network = await networks[l1ChainId]
-   const inboxAddress = await l1Network.ethBridge.inbox;
   /**
-   * Use wallets to create an arb-ts bridge instance
-   * We'll use bridge for its convenience methods around depositing ETH to L2
+   * Use l2Network to create an arb-ts EthBridger instance
+   * We'll use EthBridger for its convenience methods around transferring ETH to L2
    */
-  const bridge = await Bridge.init(l1Wallet, l2Wallet)
+
+  const l2Network = await getL2Network(l2Provider)
+  const ethBridger = new EthBridger(l2Network)
 
   /**
    * First, let's check the l2Wallet initial ETH balance
    */
-  const l2WalletInitialEthBalance = await bridge.getL2EthBalance()
+  const l2WalletInitialEthBalance = await l2Wallet.getBalance()
 
   /**
-   * Deposit ether from L1 to L2
+   * transfer ether from L1 to L2
    * This convenience method automatically queries for the retryable's max submission cost and forwards the appropriate amount to L2
+   * Arguments required are: 
+   * (1) amount: The amount of ETH to be transferred to L2
+   * (2) l1Signer: The L1 address transferring ETH to L2
+   * (3) l2Provider: An l2 provider
    */
-  const depositTx = await bridge.depositETH(ethToL2DepositAmount)
-  const rec = await depositTx.wait()
-  console.warn('deposit L1 receipt is:', rec.transactionHash)
 
+  const depositTx = await ethBridger.deposit({
+    amount: ethToL2DepositAmount,
+    l1Signer: l1Wallet,
+    l2Provider: l2Provider
+  })
+
+  const depositRec = await depositTx.wait()
+  console.warn('deposit L1 receipt is:', depositRec.transactionHash)
+  
   /**
    * With the transaction confirmed on L1, we now wait and check for the L2 side (i.e., balance credited to L2) to be confirmed as well.
-   * First, we get our txn's sequence number from the event logs (using a handy utility method)
-   * This sequence number uniquely identifies our L1-to-L2 message:
-   */
-  const seqNumArr = await bridge.getInboxSeqNumFromContractTransaction(rec)
-  if (seqNumArr === undefined) {
-    throw new Error('no seq num')
-  }
-  console.log('inbox sequence number is found!')
-  /**
-   * Note that a single txn could (in theory) trigger many l1-to-l2 messages; we know ours only triggered 1 tho.
+   * First, we get our our L1-to-L2 message
    */
-  const seqNum = seqNumArr[0]
 
-  /**
-   * Using the sequence number, we can deterministically predict what its corresponding L2 txn will be
-   */
-  const l2TxHash = await bridge.calculateL2TransactionHash(seqNum)
-  console.log('l2TxHash is: ' + l2TxHash)
+  const l1ToL2Msg = await depositRec.getL1ToL2Message(l2Provider)
```

Comment:
we can instead `waitForL2` as I mentioned in a previous comment

Reply:
updated